service/frontend/adminHandler.go

```diff
@@ -1810,3 +1820,30 @@ func convertFilterListToMap(filters []*types.DynamicConfigFilter) (map[dc.Filter
 	}
 	return newFilters, nil
 }
+
+// Ensures that shards received are within the range of the server,
+// or they'll be filtered out. This necessarily means that replication requests
+// that are for shards that don't exist are dropped.
+//
+// Replication is therefore the min(clusterA.NumberHistoryShards, clusterB.NumberHistoryShards).
```

Comment:
It's not clear to me how the shard mapping is done. 
Let's say 
- cluster A has 8k shards and cluster B has 16k shards.
- hash(wf_id_1) % 16k = 1 
- hash(wf_id_1) % 8k = 3

wf_id_1 maps to different shards due to modulo operation. do we handle re-mapping from shard 1 to 3 on the receiver cluster (B)?

Reply:
yes, the cadence history DB shard hashing tales the polls and successfully inserts it with the correct shard on the new cluster after pulling the data. I have tested this locally. The insertion is not done into the wrong shard in the DB. replicated worklflows appear to have history and are describeable 

Am I 100% certain this is bug free: no, but it seems to work as expected.

Reply:
I see. thanks for validating this locally.